moveit_core/CMakeLists.txt

```diff
@@ -10,60 +13,76 @@ add_compile_options(-Wall -Wextra
 endif()
 
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+# if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
+#   message("${PROJECT_NAME}: You did not request a specific build type: Choosing 'Release' for maximum performance")
+#   set(CMAKE_BUILD_TYPE Release)
+# endif()
 
-if(NOT CMAKE_CONFIGURATION_TYPES AND NOT CMAKE_BUILD_TYPE)
-  message("${PROJECT_NAME}: You did not request a specific build type: Choosing 'Release' for maximum performance")
-  set(CMAKE_BUILD_TYPE Release)
-endif()
-
+find_package(ament_cmake REQUIRED)
+find_package(rclcpp REQUIRED)
 find_package(Boost REQUIRED system filesystem date_time thread iostreams)
-find_package(Eigen3 REQUIRED)
 
+find_package(Eigen3 REQUIRED)
 find_package(PkgConfig REQUIRED)
-pkg_check_modules(LIBFCL_PC REQUIRED fcl)
-# find *absolute* paths to LIBFCL_INCLUDE_DIRS and LIBFCL_LIBRARIES
-find_path(LIBFCL_INCLUDE_DIRS fcl/config.h HINTS ${LIBFCL_PC_INCLUDE_DIR} ${LIBFCL_PC_INCLUDE_DIRS})
-find_library(LIBFCL_LIBRARIES fcl HINTS ${LIBFCL_PC_LIBRARY_DIRS})
+
+pkg_check_modules(LIBFCL REQUIRED fcl)
+# replace LIBFCL_LIBRARIES with full path to the library
+find_library(LIBFCL_LIBRARIES_FULL ${LIBFCL_LIBRARIES} ${LIBFCL_LIBRARY_DIRS})
+set(LIBFCL_LIBRARIES "${LIBFCL_LIBRARIES_FULL}")
 
 find_package(octomap REQUIRED)
 find_package(urdfdom REQUIRED)
+find_package(urdf REQUIRED)
 find_package(urdfdom_headers REQUIRED)
-
-find_package(catkin REQUIRED
-COMPONENTS
-  tf2_eigen
-  tf2_geometry_msgs
-  eigen_stl_containers
-  geometric_shapes
-  geometry_msgs
-  kdl_parser
-  moveit_msgs
-  octomap_msgs
-  random_numbers
-  roslib
-  rostime
-  rosconsole
-  sensor_msgs
-  shape_msgs
-  srdfdom
-  std_msgs
-  trajectory_msgs
-  urdf
-  visualization_msgs
-  xmlrpcpp
-)
-
-set(VERSION_FILE_PATH "${CATKIN_DEVEL_PREFIX}/include")
+find_package(ament_cmake REQUIRED)
+find_package(tf2_eigen REQUIRED)
+find_package(tf2_kdl REQUIRED)
+find_package(tf2_geometry_msgs REQUIRED)
+find_package(eigen_stl_containers REQUIRED)
+find_package(geometric_shapes REQUIRED)
+find_package(geometry_msgs REQUIRED)
+find_package(kdl_parser REQUIRED)
+find_package(moveit_msgs REQUIRED)
+find_package(octomap_msgs REQUIRED)
+find_package(random_numbers REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(shape_msgs REQUIRED)
+find_package(srdfdom REQUIRED)
+find_package(std_msgs REQUIRED)
+find_package(trajectory_msgs REQUIRED)
+find_package(visualization_msgs REQUIRED)
+find_package(common_interfaces REQUIRED)
+find_package(rcutils REQUIRED)
+find_package(pluginlib REQUIRED)
+
+set(VERSION_FILE_PATH ${CMAKE_BINARY_DIR}/include)
 # Pass the folder of the generated version.h to catkin_package() for export in devel-space
 # This is how gencpp adds the folder of generated message code to the include dirs, see:
 #   https://github.com/ros/gencpp/blob/e5acaf6/cmake/gencpp-extras.cmake.em#L51-L54
 list(APPEND ${PROJECT_NAME}_INCLUDE_DIRS ${VERSION_FILE_PATH})
+# Create a directory at build/moveit_core/include/moveit
```

Comment:
Add to comment why you are making it.. the command itself already basically  says what it is doing